apps/prairielearn/src/sync/course-db.ts

```diff
@@ -968,8 +992,8 @@ async function validateAssessment(
   // which are accessible either now or any time in the future.
   if (!courseInstanceExpired) {
     (assessment.allowAccess || []).forEach((rule) => {
-      const allowAccessWarnings = checkAllowAccessRoles(rule);
-      warnings.push(...allowAccessWarnings);
+      warnings.push(...checkAllowAccessRoles(rule));
+      warnings.push(...checkAllowAccessUids(rule));
```

Comment:
As with most new validations we add, we're only doing this check in the context of course instances that are either currently accessible or will be accessible in the future.